src/main/java/fi/fmi/avi/model/swx/immutable/SpaceWeatherAdvisoryImpl.java

```diff
@@ -105,6 +118,62 @@ public Builder addAllPhenomena(final List<SpaceWeatherPhenomenon> elements) {
             return super.addAllPhenomena(elements);
         }
 
+        public SpaceWeatherAdvisoryImpl.Builder withCompleteIssueTimeNear(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            return mapIssueTime((input) -> input.toBuilder().completePartialNear(reference).build());
+        }
+
+        private SpaceWeatherAdvisoryImpl.Builder withCompleteNextAdvisory(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            return mapNextAdvisory(nextAdvisory -> {
+                final NextAdvisoryImpl.Builder builder = NextAdvisoryImpl.Builder.from(nextAdvisory);
+                builder.mapTime(time -> time.toBuilder().completePartialNear(reference).build());
+                return builder.build();
+            });
+        }
+
+        private Builder withCompleteAnalysisTimes(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            if (!getAnalyses().isEmpty()) {
+                final Iterable<PartialOrCompleteTimeInstant> partialTimes = getAnalyses().stream().map(SpaceWeatherAdvisoryAnalysis::getTime)::iterator;
```

Comment:
Do note that this iterator iterates only once. To allow multiple iterations, this should be `() -> getAnalyses().xxx().iterator()`. In this case it may not matter, but just in case...